src/config-file.c

```diff
@@ -222,28 +222,38 @@ static gboolean get_group(GKeyFile *key_file, const gchar *group,
         return TRUE;
 }
 
+/**
+ * @brief Get GLogLevelFlags for error string.
+ *
+ * @param[in]  log_level Log level string
+ * @return GLogLevelFlags matching error string, else default log level
+ *         (message)
+ */
 static GLogLevelFlags log_level_from_string(const gchar *log_level)
 {
-        if (g_strcmp0(log_level, "error") == 0) {
+        g_return_val_if_fail(log_level, 0);
+
+        if (g_strcmp0(log_level, "error") == 0)
                 return G_LOG_LEVEL_ERROR;
-        } else if (g_strcmp0(log_level, "critical") == 0) {
+        else if (g_strcmp0(log_level, "critical") == 0)
                 return G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL;
-        } else if (g_strcmp0(log_level, "warning") == 0) {
+        else if (g_strcmp0(log_level, "warning") == 0)
                 return G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL |
                        G_LOG_LEVEL_WARNING;
-        } else if (g_strcmp0(log_level, "message") == 0) {
+        else if (g_strcmp0(log_level, "message") == 0)
                 return G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL |
                        G_LOG_LEVEL_WARNING | G_LOG_LEVEL_MESSAGE;
-        } else if (g_strcmp0(log_level, "info") == 0) {
+        else if (g_strcmp0(log_level, "info") == 0)
                 return G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL |
                        G_LOG_LEVEL_WARNING | G_LOG_LEVEL_MESSAGE |
                        G_LOG_LEVEL_INFO;
-        } else if (g_strcmp0(log_level, "debug") == 0) {
-                return G_LOG_LEVEL_MASK;
-        } else {
+        else if (g_strcmp0(log_level, "debug") == 0)
+                return G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL |
+                       G_LOG_LEVEL_WARNING | G_LOG_LEVEL_MESSAGE |
+                       G_LOG_LEVEL_INFO | G_LOG_LEVEL_DEBUG;
+        else
                 return G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL |
```

Comment:
Maybe this is a case for a `g_error()` or at least `g_warning`. If we passed an invalid 'log_level' we should not silently accept this as this is a coding error.

Reply:
The value comes from the config file, so it is a configuration error. I've added a `g_warning`.